src/components/edit/FacilityDialog.js

```diff
@@ -39,234 +39,182 @@ import BufferRadius from './shared/BufferRadius.js'
-    static propTypes = {
-        setOrgUnitGroups: PropTypes.func.isRequired,
-        setOrgUnitLevels: PropTypes.func.isRequired,
-        setOrganisationUnitColor: PropTypes.func.isRequired,
-        setOrganisationUnitGroupSet: PropTypes.func.isRequired,
-        setRadiusLow: PropTypes.func.isRequired,
-        setUserOrgUnits: PropTypes.func.isRequired,
-        toggleOrgUnit: PropTypes.func.isRequired,
-        validateLayer: PropTypes.bool.isRequired,
-        onLayerValidation: PropTypes.func.isRequired,
-        id: PropTypes.string,
-        orgUnitField: PropTypes.string,
-        organisationUnitColor: PropTypes.string,
-        organisationUnitGroupSet: PropTypes.object,
-        radiusLow: PropTypes.number,
-        rows: PropTypes.array,
-    }
+const QUERY = {
+    configuration: {
+        resource: 'configuration',
+    },
+}
 
-    constructor(props, context) {
-        super(props, context)
-        this.state = {
-            tab: 'orgunits',
+const ORGUNITS_TAB = 'orgunits'
+
+const FacilityDialog = ({
+    rows = [],
+    radiusLow,
+    organisationUnitColor,
+    organisationUnitGroupSet,
+    orgUnitField,
+    id,
+    validateLayer,
+    onLayerValidation,
+}) => {
+    const [tab, setTab] = useState(ORGUNITS_TAB)
+    const [orgUnitsError, setOrgUnitsError] = useState()
+    const { data } = useDataQuery(QUERY)
+    const dispatch = useDispatch()
+
+    const facilityOrgUnitLevel = data?.configuration.facilityOrgUnitLevel
+    const facilityOrgUnitGroupSet = data?.configuration.facilityOrgUnitGroupSet
+
+    const orgUnits = getOrgUnitsFromRows(rows)
+
+    const validate = useCallback(() => {
+        if (!orgUnits.length) {
+            setTab(ORGUNITS_TAB)
+            setOrgUnitsError(i18n.t('No organisation units are selected'))
+            return false
         }
-    }
-
-    componentDidMount() {
-        fetchFacilityConfigurations().then((config) => this.setState(config))
-    }
 
-    componentDidUpdate(prevProps, prevState) {
-        const {
-            id,
-            rows,
-            setOrgUnitLevels,
-            organisationUnitGroupSet,
-            setOrganisationUnitGroupSet,
-            validateLayer,
-            onLayerValidation,
-        } = this.props
+        return true
+    }, [orgUnits])
 
-        const { facilityOrgUnitLevel, facilityOrgUnitGroupSet } = this.state
+    useEffect(() => {
+        if (validateLayer) {
+            onLayerValidation(validate())
+        }
+    }, [validateLayer, onLayerValidation, validate])
 
-        // If new layer
+    useEffect(() => {
         if (!id) {
             // Set default org unit level
-            if (
-                !getOrgUnitsFromRows(rows).length &&
-                facilityOrgUnitLevel &&
-                !prevState.facilityOrgUnitLevel
-            ) {
-                setOrgUnitLevels([facilityOrgUnitLevel.id])
+            if (facilityOrgUnitLevel) {
+                dispatch(setOrgUnitLevels([facilityOrgUnitLevel.id]))
```

Comment:
I think this should only be dispatched if there is no `getOrgUnitsFromRows(rows)`. If the user is editing am existing map, we shouldn't change the layer config.

Reply:
The check for id on line 87 should control for that right? An existing map will have ids for the layer and thus never run this code.  Also once you create a layer on a new map, it will also have an id, so it won't run in this case either.

 Unless you know of cases where there might not be an id